recipes/catch2/2.x.x/conanfile.py

```diff
@@ -30,109 +32,112 @@ class Catch2Conan(ConanFile):
-            raise ConanInvalidConfiguration("ARMv8 is supported by 2.13.1+ only! give up!")
-        if self.options.with_main and tools.Version(self.version) < "2.13.4":
-            raise ConanInvalidConfiguration("Option with_main not supported with versions < 2.13.4")
+        if Version(self.version) < "2.13.1" and self.settings.arch == "armv8":
+            raise ConanInvalidConfiguration("ARMv8 is not supported by versions < 2.13.1+")
+        if self.options.with_main and Version(self.version) < "2.13.4":
+            raise ConanInvalidConfiguration("Option with_main not supported by versions < 2.13.4")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
-
-    @functools.lru_cache(1)
-    def _configure_cmake(self):
-        cmake = CMake(self)
-        cmake.definitions["BUILD_TESTING"] = "OFF"
-        cmake.definitions["CATCH_INSTALL_DOCS"] = "OFF"
-        cmake.definitions["CATCH_INSTALL_HELPERS"] = "ON"
-        cmake.definitions["CATCH_BUILD_STATIC_LIBRARY"] = self.options.with_main
-        cmake.definitions["enable_benchmark"] = self.options.get_safe("with_benchmark", False)
-        cmake.definitions["CATCH_CONFIG_PREFIX_ALL"] = self.options.with_prefix
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["BUILD_TESTING"] = False
+        tc.cache_variables["CATCH_INSTALL_DOCS"] = False    # these are cmake options, so use cache_variables
+        tc.cache_variables["CATCH_INSTALL_HELPERS"] = "ON"  # these are cmake options, so use cache_variables
+        tc.cache_variables["CATCH_BUILD_STATIC_LIBRARY"] = str(self.options.with_main)   # these are cmake options, so use cache_variables (str() is required for conan 1.52)
+        if self.options.with_prefix:
+            tc.preprocessor_definitions["CATCH_CONFIG_PREFIX_ALL"] = 1
+        if self.options.get_safe("with_benchmark", False):
+            tc.preprocessor_definitions["CATCH_CONFIG_ENABLE_BENCHMARKING"] = 1
         if self.options.default_reporter:
-            cmake.definitions["CATCH_CONFIG_DEFAULT_REPORTER"] = self._default_reporter_str
-
-        cmake.configure(build_folder=self._build_subfolder)
-        return cmake
+            tc.variables["CATCH_CONFIG_DEFAULT_REPORTER"] = self._default_reporter_str
+        tc.generate()
 
     def build(self):
         # Catch2 does skip install if included as subproject:
         # https://github.com/catchorg/Catch2/blob/79a5cd795c387e2da58c13e9dcbfd9ea7a2cfb30/CMakeLists.txt#L100-L102
-        main_cml = os.path.join(self._source_subfolder, "CMakeLists.txt")
-        tools.replace_in_file(main_cml, "if (NOT_SUBPROJECT)", "if (TRUE)")
+        main_cml = os.path.join(self.source_folder, "CMakeLists.txt")
+        replace_in_file(self, main_cml, "if (NOT_SUBPROJECT)", "if (TRUE)")
```

Comment:
These could be removed if we drop the root one in the recipe

Reply:
I love deleting lines